frontend/public/components/utils/category-filter-utils.ts

```diff
@@ -0,0 +1,391 @@
+  }
+
+  return updatedFilters;
+};
+
+const clearActiveFilters = (activeFilters, filterGroups) => {
+  const clearedFilters = _.cloneDeep(activeFilters);
+
+  // Clear the keyword filter
+  _.set(clearedFilters, 'keyword.value', '');
+  _.set(clearedFilters, 'keyword.active', false);
+
+  // Clear the group filters
+  _.each(filterGroups, field => {
+    _.each(_.keys(clearedFilters[field]), key => _.set(clearedFilters, [field, key, 'active'], false));
+  });
+
+  return clearedFilters;
+};
+
+const setURLParams = params => {
+  const location: any = window.location;
+  const url = new URL(location);
+  const searchParams = `?${params.toString()}${url.hash}`;
+
+  history.replace(`${url.pathname}${searchParams}`);
+};
+
+const updateURLParams = (filterName, value) => {
+  const params = new URLSearchParams(window.location.search);
+
+  if (value) {
+    params.set(filterName, Array.isArray(value) ? JSON.stringify(value) : value);
+  } else {
+    params.delete(filterName);
+  }
+  setURLParams(params);
+};
+
+const clearFilterURLParams = selectedCategoryId => {
+  const params = new URLSearchParams();
+
+  if (selectedCategoryId) {
+    params.set(CATEGORY_URL_PARAM, selectedCategoryId);
+  }
+
+  setURLParams(params);
+};
+
+const getURLParam = param => {
+  const params = new URLSearchParams(window.location.search);
+  return params.get(param);
+};
+
+const getActiveValuesFromURL = (availableFilters, filterGroups) => {
+  const searchParams = new URLSearchParams(window.location.search);
+  const categoryParam = searchParams.get(CATEGORY_URL_PARAM);
+  let keywordFilter = searchParams.get(KEYWORD_URL_PARAM);
+
+  const selectedCategoryId = categoryParam || 'all';
+
+  const groupFilters = {};
+
+  _.forEach(filterGroups, filterGroup => {
+    const groupFilterParam = searchParams.get(filterGroup);
+    if (groupFilterParam) {
+      try {
+        _.set(groupFilters, [filterGroup], JSON.parse(groupFilterParam));
+      } catch (e) {
+        // eslint-disable-next-line no-console
+        console.warn('could not update filters from url params: could not parse search params');
+      }
+    }
+  });
+
+  let activeFilters = getActiveFilters(keywordFilter, groupFilters, availableFilters);
+
+  return {selectedCategoryId, activeFilters};
+};
+
+const getFilterSearchParam = groupFilter => {
+  let activeValues = [];
+  _.each(Object.keys(groupFilter), typeKey => {
+    if (groupFilter[typeKey].active) {
+      activeValues.push(typeKey);
+    }
+  });
+
+  if (_.size(activeValues)) {
+    return JSON.stringify(activeValues);
+  }
+
+  return '';
+};
+
+const getFilterGroupCounts = (items, filterGroups, selectedCategoryId, filters, categories) => {
+  const categoriesForCounts = recategorizeItems(items, categories);
+
+  const activeCategory = findActiveCategory(selectedCategoryId, categoriesForCounts);
+  const activeItems = activeCategory ? activeCategory.items : 0;
+  const newFilterCounts = {};
+
+  _.forEach(filterGroups, filterGroup => {
```

Comment:
style nit: consistently use either `_.forEach` or `_.each`. There's a mix here. We usually use `_.each`